apps/core/declarativewebcontainer.cpp

```diff
@@ -69,6 +71,7 @@ DeclarativeWebContainer::DeclarativeWebContainer(QWindow *parent)
     , m_activeTabRendered(false)
     , m_clearSurfaceTask(0)
     , m_closing(false)
+    , m_touchScreenDevice(new QTouchDevice)
```

Comment:
Newer destroyed? Could this be on stack instead?

Reply:
Yikes. QTouchDevice is Q_GADGET and not QObject so added manual delete. Postponed the initialization to receiving of first mouse events so should not affect touch-driven browser usage.